File: AI.Core/OperatorsImplimentation/BasicOperators/Division.cs
namespace AI.Core.OperatorsImplimentation.BasicOperators
{
    class Division : Operator
    {
        public Division(Entity thisEntity)
            : base(thisEntity)
        {
            EnterContactsCount = 2;
        }

        public override void Action()
        {
            var value = EnterContacts[0].Value/EnterContacts[1].Value;
            ExitContacts.SetValue(value);
        }
    }
}

File: NativeCore/Division.cpp
#include "Division.h"
#include "stdafx.h"

namespace AI
{
	namespace Core
	{
		namespace OperatorsImplimentation
		{
			namespace BasicOperators
			{

				Division::Division(Entity *thisEntity) : Operator(thisEntity)
				{
					setEnterContactsCount(2);
				}

				void Division::Action()
				{
					auto value = EnterContacts[0]->Value / EnterContacts[1]->Value;
					ExitContacts->SetValue(value);
				}
			}
		}
	}
}
